### ISSUE TITLE
Remove unnecessary CA2022 suppressions

### ISSUE THREAD

**Issue by NikolaMilosavljevic** [dotnet/source-build#4322]
As part of re-bootstrapping in https://github.com/dotnet/installer/pull/19145 we've encountered numerous CA2022 errors. This required patches for several repos. Most of those changes were backported.

It appears that compiler/analyzer was unnecessarily restrictive and the change was made in https://github.com/dotnet/roslyn-analyzers/issues/7268 to relax the check.

Once we pick up newer SDK (`9.0.100-preview.4.24209.7` or later) we should be able to remove all of these suppressions and backport them.

**Comment by MichaelSimons** [comment]
[Triage] Is there anymore work to do here? The last patches are being removed in https://github.com/dotnet/installer/pull/19508.

**Comment by NikolaMilosavljevic** [comment]
This issue tracks removal of suppressions that were added with https://github.com/dotnet/installer/pull/19145

Analyzer was too restrictive - it was fixed with https://github.com/dotnet/roslyn-analyzers/issues/7268

This work can be done after next VMR re-bootstrapping.

**Comment by MichaelSimons** [comment]
@NikolaMilosavljevic, is this something @Winniexu01 could help out with?  If so please provide guidance.

**Comment by NikolaMilosavljevic** [comment]
> [@NikolaMilosavljevic](https://github.com/NikolaMilosavljevic), is this something [@Winniexu01](https://github.com/Winniexu01) could help out with? If so please provide guidance.

Yes, we should be able to remove all of these suppressions. The code might require some minor tweaks.
